view_pandas.py

```diff
@@ -0,0 +1,128 @@
+                            title=title)
+            plot.add_tools(PanTool(),
+                           WheelZoomTool(),
+                           BoxSelectTool(),
+                           HoverTool())
+        self.plot = plot
+
+
+class View(object):
+    """Puts information from Model() on top of the nice gmap it got from MakePlot().
+    """
+
+    def __init__(self, filename='test.csv', model=None, plot=None, button=None):
+        self.filename = filename
+        if model is None:
+            self.model = mdl.Model(self.filename)
+        else:
+            self.model = model
+        if plot is None:
+            x = MakePlot()
+            self.plot = x.plot
+        else:
+            self.plot = plot
+        if button is None:
+            button = CheckboxButtonGroup(labels=['were trying', 'did we succeed'])
+            self.button = button
+        else:
+            self.button = button
+
+    def marker(self):
+        """Defines a marker (a circle glyph) to represent a resource on the map.
+        """
+        self.model.set_color()
+        circle = Circle(x='lon', y='lat', size=15, fill_color='color', fill_alpha=.8)
+        self.plot.add_glyph(self.model.source, circle)
+
+    def legend(self):
+        """Defines and adds to plot a legend that describes what type of
+        resource each color glyph maps to.
+
+        We did not have time to fully implement this before the due date.
+        The legend object would appear but would be empty.
+        """
+        """
+        red_glyph = self.plot.add_glyph(self.model.source, Circle(x='lon', y='lat', size=15, fill_color='red', fill_alpha=.8))
+        Legend.legends = [
+            ('health', red_glyph)]
+            ('support', green_glyph),
+            ('housing', blue_glyph),
+            ('advocacy', purple_glyph),
+            ('legal', cyan_glyph)]
+        legend = Legend(location='bottom_left')
+        self.plot.add_layout(legend)
+        """
+        pass  # TODO: Implement this
+
+    def hover_tool(self):
+        """Makes the hover tool! Woot!
+        """
+        hover = self.plot.select_one(HoverTool)
+        hover.point_policy = 'follow_mouse'
+        hover.tooltips = [('Name', '@name'),
+                          ('Address', '@address'),
+                          ('Type', '@category')]
+
+    def buttons(self):
+        """Makes buttons appear.
+
+        Right now we're struggling with this because we probably have to use
+        either a server or callbacks, one of which is being difficult and also
+        one that involves a JS snippet.
+        """
+        pass  # TODO: Implement this.
+
+    def show_plot(self):
+        """Outputs and displays a plot on a static webpage.
+        """
+        self.marker()
+        self.hover_tool()
+#        self.legend()
+        self.buttons()
+        output_file('resources_plot.html')
+        show(self.plot)
+
+
+if __name__ == '__main__':
+    thing = View()
```

Comment:
Overall - code looks good! It was nicely designed and well documented. For the most part, variable names were sensibly chosen.

You should include a .gitignore file in your repository to prevent *.pyc files from being pushed. For more info on .gitignore, see https://git-scm.com/docs/gitignore